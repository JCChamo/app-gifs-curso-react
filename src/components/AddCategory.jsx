import { useState } from 'react';
import { PropTypes } from 'prop-types';

export function AddCategory({
  // setCategories
  onNewCategory,
}) {
  const [inputValue, setinputValue] = useState('');
  const onInputChange = ({ target }) => {
    setinputValue(target.value);
  };
  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;
    // setCategories((categories) => [...categories, inputValue]);
    onNewCategory(inputValue.trim());
    setinputValue('');
    // console.log(inputValue);
  };
  return (
    <form onSubmit={onSubmit} aria-label='form'>
      <input
        type='text'
        placeholder='Buscar gifs'
        value={inputValue}
        onChange={(event) => onInputChange(event)}
      />
    </form>
  );
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
