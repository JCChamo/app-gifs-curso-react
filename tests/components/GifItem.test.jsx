import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components';

describe('Pruebas de <GifItem/>', () => {
  const title = 'patata';
  const url = 'https://patata.gif/';
  test('Debe coincidir con el snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
  test('debería mostrar imagen con url y alt indicado', () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });
  test('debe existir el título', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
  // test("Comprobación de parámetros obligatorios de props title y url", () => {});
});
