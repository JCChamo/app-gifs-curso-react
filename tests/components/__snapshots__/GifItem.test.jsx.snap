// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifItem/> Debe coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="patata"
      src="https://patata.gif/"
    />
    <p>
      patata
    </p>
  </div>
</div>
`;
